lib/workers/repository/reconfigure/index.ts

```diff
@@ -17,15 +33,114 @@ export async function checkReconfigureBranch(
     return;
   }
 
+  const context = config.statusCheckNames?.configValidation;
   const reconfigureBranch = getReconfigureBranchName(config.branchPrefix!);
   const branchExists = await scm.branchExists(reconfigureBranch);
 
-  // this is something the user initiates, so skip if no branch exists
+  // this is something the user initiates, skip if no branch exists
   if (!branchExists) {
     logger.debug('No reconfigure branch found');
     deleteReconfigureBranchCache(); // in order to remove cache when the branch has been deleted
     return;
   }
 
-  await validateReconfigureBranch(config);
+  const existingPr = await platform.findPr({
+    branchName: reconfigureBranch,
+    state: 'open',
+    includeOtherAuthors: true,
+    targetBranch: config.defaultBranch,
+  });
+  const branchSha = getBranchCommit(reconfigureBranch)!;
+  const cache = getCache();
+  const reconfigureCache = cache.reconfigureBranchCache;
+
+  // only use valid cached information
+  if (reconfigureCache?.reconfigureBranchSha === branchSha) {
+    if (!existingPr || reconfigureCache.extractResult) {
+      logger.debug('Skipping validation check as branch sha is unchanged');
+      return;
+    }
+  }
+
+  const result = await getReconfigureConfig(reconfigureBranch);
+
+  if (!result.ok) {
+    await setBranchStatus(reconfigureBranch, result.errMessage, 'red', context);
+    setReconfigureBranchCache(branchSha, false);
+    await scm.checkoutBranch(config.defaultBranch!);
+    return;
+  }
+
+  const isValidConfig = await validateReconfigureBranch(
+    config,
+    result.config,
+    result.configFileName,
+    existingPr,
+  );
+
+  if (!isValidConfig) {
+    logger.debug(
+      'Found errors in reconfigure config. Skipping dependencies extraction',
+    );
+    await scm.checkoutBranch(config.defaultBranch!);
+    return;
+  }
+
+  if (!existingPr) {
+    logger.debug('No reconfigure pr found. Skipping dependencies extraction');
+    await scm.checkoutBranch(config.defaultBranch!);
+    return;
+  }
+
+  let extractResult: ExtractResult | undefined;
+  // Recompute config similar to repo config processing
+  // Get non-global config from file config
+  // Merge it with inherited and static repo config
+  // Finally, merge the reconfigure config
+  let newConfig = GlobalConfig.set(repoConfig);
```

Comment:
do we use global config anywhere after the reconfigure thing? it could contain wrong / unexpected values then.

ok new config can also contain new hostrules 🤔

Reply:
We are just resetting it to the same values so it should not have any effect